File: generated-schemas/samples/v2/Samples.java
package org.observertc.webrtc.observer.samples;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
/** 
* A compound object hold samples and control flags
*/
@JsonIgnoreProperties(ignoreUnknown = true)
public  class Samples { 

	/**
	* Additional meta information about the carried payloads
	*/
	@JsonProperty("meta")
	public SamplesMeta meta;
	
	/**
	* array of client samples
	*/
	@JsonProperty("clientSamples")
	public ClientSample[] clientSamples;
	
	/**
	* array of sfu samples
	*/
	@JsonProperty("sfuSamples")
	public SfuSample[] sfuSamples;
	
	/**
	* Additional control flags indicate various operation has to be performed
	*/
	@JsonProperty("controlFlags")
	public ControlFlags controlFlags;
	
}

File: sources/samples/Samples.ts
// @revision: 1

import { ClientSample } from "./ClientSample";
import { SfuSample } from "./SfuSample";

/**
 * A compound object hold samples and control flags
 */
export interface Samples {
    /**
     * array of client samples
     */
    clientSamples?: ClientSample[];

    /**
     * array of sfu samples
     */
    sfuSamples?: SfuSample[];

    // control flags
}